charmhelpers/contrib/storage/linux/ceph.py

```diff
@@ -614,7 +614,7 @@ def create(self):
 
 class ReplicatedPool(BasePool):
     def __init__(self, service, name=None, pg_num=None, replicas=None,
-                 percent_data=None, app_name=None, op=None):
+                 percent_data=None, app_name=None, profile_name=None, op=None):
```

Comment:
I'd rather the new argument (`profile_name`) be added to the end of this list, and also to have the function signature include the default:

```
    def __init__(self, service, name=None, pg_num=None, replicas=None,
                 percent_data=None, app_name=None, op=None, profile_name='replicated_rule'):
```

Is there any strong reason why it was done the other way?

Reply:
None, I thought this was already covered with line 656

Reply:
> I'd rather the new argument (profile_name) be added to the end of this list, and also to have the function signature include the default:

With named params the order generally doesn't matter, and so I let that one through.

re: default.  Personally, I **much** prefer people to signal 'None' as a default parameter as the type can then be `Optional[str]` rather than `str` (which is a good signal that it is optional).  By setting the default if None is passed, it's really obvious in the code that the default is being used.  Finally, it stops (forever) the risk that someone will accidentally put an internally-mutable reference as a default leading to all sorts of shenanigans.

e.g.

```python
def some_function(arg=[]):
    arg.append("thing")
    print(arg)
```

:)

So, I'd like optional arguments to still be `None`.

Reply:
I'm very aware of mutable references as defaults; _however_, this is a `str` and, while is has a default, it isn't really optional after this change. It's entirely OK to leave it as the default but a `None`, in this case, isn't actually true. The final resting place of this argument looks like:

```python
cmd = [
    'ceph', '--id', self.service, 'osd', 'pool', 'create',
    self.name, str(self.pg_num), self.profile_name
]
```

Given that the `profile_name` isn't actually optional, I'd prefer if the type signature actually tells me what the default is rather than having to read the function's docstrings to figure out what the default profile name is.